### PR TITLE
o2-eve: implementation of *.eve binary format, many fixes

### PR DESCRIPTION
**Pull request by jmyrcha**
*.eve binary format allows faster serialization of data and is compatible with web client

### CODE REVIEW

File: EventVisualisation/DataConverter/include/EventVisualisationDataConverter/VisualisationEvent.h [hunk 1]
```diff
@@ -66,7 +68,7 @@ class VisualisationEvent
     time_t collisionTime;
   };
   // Default constructor
-  VisualisationEvent(const VisualisationEventVO vo);
+  explicit VisualisationEvent(VisualisationEventVO vo);
```
Comment (shahor02) [inline EventVisualisation/DataConverter/include/EventVisualisationDataConverter/VisualisationEvent.h]: Should this have been passed by const reference ?

File: EventVisualisation/DataConverter/include/EventVisualisationDataConverter/VisualisationEvent.h [hunk 2]
```diff
@@ -75,23 +77,23 @@ class VisualisationEvent
...
-  VisualisationCluster& addCluster(float XYZ[], float trackTime)
+  VisualisationCluster& addCluster(float X, float Y, float Z)
   {
-    return mTracks.back().addCluster(XYZ);
+    float pos[] = {X, Y, Z};
+    return mTracks.back().addCluster(pos);
   }
 
-  VisualisationCluster& addCluster(float X, float Y, float Z, float trackTime)
+  void addGlobalCluster(float* xyz, float time, o2::dataformats::GlobalTrackID gid)
```
Comment (shahor02) [inline EventVisualisation/DataConverter/include/EventVisualisationDataConverter/VisualisationEvent.h:87]: `const float* xyz` ?

File: EventVisualisation/DataConverter/src/VisualisationCluster.cxx [hunk 3]
```diff
@@ -24,10 +24,11 @@ namespace o2
 namespace event_visualisation
 {
 
-VisualisationCluster::VisualisationCluster(float XYZ[], float time)
+VisualisationCluster::VisualisationCluster(float XYZ[], float time, o2::dataformats::GlobalTrackID gid)
```
Comment (shahor02) [inline EventVisualisation/DataConverter/src/VisualisationCluster.cxx]: `const float XYZ[]`?

File: EventVisualisation/DataConverter/src/VisualisationCluster.cxx [hunk 4]
```diff
@@ -24,10 +24,11 @@ namespace o2
...
 {
   setCoordinates(XYZ);
   this->mTime = time;
+  this->mBGID = gid;
 }
 
 void VisualisationCluster::setCoordinates(float xyz[3])
```
Comment (shahor02) [inline EventVisualisation/DataConverter/src/VisualisationCluster.cxx]: same here

File: EventVisualisation/DataConverter/include/EventVisualisationDataConverter/VisualisationEvent.h (hunk 1 above)
Comment (jmyrcha) [reply EventVisualisation/DataConverter/include/EventVisualisationDataConverter/VisualisationEvent.h]: done

File: EventVisualisation/DataConverter/src/VisualisationCluster.cxx (hunk 4 above)
Comment (jmyrcha) [reply EventVisualisation/DataConverter/src/VisualisationCluster.cxx]: done

File: EventVisualisation/DataConverter/src/VisualisationCluster.cxx (hunk 3 above)
Comment (jmyrcha) [reply EventVisualisation/DataConverter/src/VisualisationCluster.cxx]: done

File: EventVisualisation/DataConverter/include/EventVisualisationDataConverter/VisualisationEvent.h (hunk 2 above)
Comment (jmyrcha) [reply EventVisualisation/DataConverter/include/EventVisualisationDataConverter/VisualisationEvent.h:87]: done

**Pull request merged**